src/coreComponents/physicsSolvers/contact/LagrangianContactSolver.cpp

```diff
@@ -539,14 +539,26 @@ void LagrangianContactSolver::setupDofs( DomainPartition const & domain,
     meshTargets[std::make_pair( meshBodyName, meshLevel.getName())] = std::move( regions );
   } );
 
+  dofManager.addField( solidMechanics::totalDisplacement::key(),
+                       FieldLocation::Node,
+                       3,
```

Comment:
We decoupled DoF definitions from physical fields on purpose to allow more flexibility in assigning/blocking dofs (e.g. node-wise vs component-wise numbering). But it should be possible to add a method that takes a Field trait and adds a DoF according to it, since this is the default use case.